From 6c3bd8844252d3bc5070dbf9c60e8945aacea141 Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Fri, 8 Jul 2016 15:20:16 +0800
Subject: [PATCH 1108/5242] MLK-12975-1 video: mipi_dsi_samsung: set default
 bpp to 32 if not set

commit  714a55c3bf2c49d3c5267ff5c5f65e270b673531 from
https://source.codeaurora.org/external/imx/linux-imx.git

The default bpp should be set to a proper value if it is
not set yet during mipi panel display initialization.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/video/fbdev/mxc/mipi_dsi_samsung.c |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/video/fbdev/mxc/mipi_dsi_samsung.c b/drivers/video/fbdev/mxc/mipi_dsi_samsung.c
index 7de2e24..491ea68 100644
--- a/drivers/video/fbdev/mxc/mipi_dsi_samsung.c
+++ b/drivers/video/fbdev/mxc/mipi_dsi_samsung.c
@@ -112,6 +112,10 @@ static int mipi_dsi_lcd_init(struct mipi_dsi_info *mipi_dsi,
 		return -EINVAL;
 	}
 
+	/* set default bpp to 32 if not set*/
+	if (!setting->default_bpp)
+		setting->default_bpp = 32;
+
 	mipi_dsi->lcd_callback->get_mipi_lcd_videomode(&mipi_lcd_modedb, &size,
 					&mipi_dsi->lcd_config);
 
-- 
1.7.9.5

